### PR TITLE
Feature: New taxonomy filters to use taxonomies in the question selection for fixed tests

### PR DESCRIPTION
**Pull request by matheuszych**
This is the implentation of the Feature Wiki entry [New taxonomy filters to use taxonomies in the question selection for fixed tests](https://docu.ilias.de/go/wiki/wpage_8158_1357).

I believe all the requested features are implemented according to the Request.

### Important!
The filtering of the data is dependent on this https://github.com/ILIAS-eLearning/ILIAS/pull/7940 PR.

### CODE REVIEW

**Comment by dsstrassner** [comment]
@thojou is this PR already reviewed by you and the customer? Please provide an "Abnahmeplattform" for me.

**Comment by thojou** [comment]
Hey @dsstrassner,

I've reviewed the PR, and I believe the implementation is solid and well-structured.

Based on the Feature-Wiki, there is a slight divergence in the current implementation that we've already discussed with @oliversamoila. Specifically, the "Question Pool Title" field in the Filter and Table has been renamed to "Parent Object Title". This change was made because the related table is context-sensitive and can be used for adding questions either from 1. another question pool or 2. another test. In the second scenario, the label "Question Pool Title" wouldn't be appropriate.

An alternative approach could be to dynamically adjust the title based on context, switching between "Question Pool Title" and "Test Title" as needed.

We've already delivered this feature to @oliversamoila and provided an acceptance platform. @oliversamoila will inform the relevant customers.

This PR is ready for further review. Perhaps @klees or @mjansenDatabay could take a look to ensure we're avoiding any potential anti-patterns.

**Comment by oliversamoila** [comment]
Hey @dsstrassner 
I'll contact you and Tomke as soon as possible.

Best regards
Oliver

**Comment by dsstrassner** [comment]
Test installation is rebased, therefor new objects will be possible. @dsstrassner will look into it, as soon as possible.

File: lang/ilias_de.lang [hunk 1]
```diff
@@ -1110,6 +1110,7 @@ assessment#:#question_is_part_of_running_test#:#Die Frage ist Teil eines laufend
 assessment#:#question_marking#:#Fragen markieren
 assessment#:#question_marking_description#:#Teilnehmer erhalten die Möglichkeit, Fragen für sich zu markieren. Die Markierungen werden auch in der 'Übersicht Testdurchlauf' angezeigt, sofern diese aktiviert ist.
 assessment#:#question_not_answered#:#Die Frage wurde nicht beantwortet
+assessment#:#question_pool_title#:#Question Pool Titel
```
Comment (oliversamoila) [inline lang/ilias_de.lang]: Sollte "Titel Fragenpool" heißen, oder?

Comment (matheuszych) [reply lang/ilias_de.lang]: Im Featurewiki steht keine Deutsche Übersetzung. Beim Englischen steht aber "Question Pool Title" weshalb "Question Pool Titel" sich, meiner Meinung nach, am besten eignet :)

Comment (dsstrassner) [reply lang/ilias_de.lang]: My fault, I did not read the Feature Request down to this level, or maybe I simply forgot to check the existing implementations. 

 

- [x] Dear @matheuszych, pease do not introduce a new label. Please use only Question Pool -- > 

assessment#:#tst_source_question_pool#:#Fragenpool
assessment#:#tst_source_question_pool#:#Question Pool

- [x] Dear @oliversamoila please add this change to the implementation informations in the Feature Wiki Article. Thanks in advance!

Comment (matheuszych) [reply lang/ilias_de.lang]: I replaced the label and the description column can now be hidden but is shown by default.

**Review by dsstrassner — COMMENTED** [review]
I tested this yesterday and today on the provided test installation. Two point is still open: 
- [ ] If I filter the table and use Bulk Actions for the entire table, I would expect that this works for the whole filtered content of the table, independent if the result has one or more table pages. And not for all the content I saw without the filtering.
- [ ] Adding optional filters are creating a error, but this seems to be a filter problem.

The other parts are ready for trunk integration from my point of view.

**Comment by thojou** [comment]
Hi @dsstrassner,

The bulk-action "For Entire Table" now respects the applied filters and executes the command on all filtered items.

As for the issue with optional filters, I've already created a Mantis ticket for it: [#42109](https://mantis.ilias.de/view.php?id=42109).

I believe we can proceed from here.

Best regards,

**Comment by matheuszych** [comment]
@kergomard 
Due to some git issues please continue to https://github.com/ILIAS-eLearning/ILIAS/pull/8264.

**Pull request closed without merging**